cmd/crank/xpkg/login.go

```diff
@@ -47,14 +47,37 @@ const (
 )
 
 type loginCmd struct {
+	// Flags. Keep sorted alphabetically.
+	Password string `short:"p" env:"UP_PASSWORD" help:"Password for specified username. '-' to read from stdin."`
+	Token    string `short:"t" env:"UP_TOKEN" xor:"identifier" help:"Token used to authenticate. '-' to read from stdin."`
+	Username string `short:"u" env:"UP_USER" xor:"identifier" help:"Username used to authenticate."`
+
+	// Common Upbound API configuration.
+	// Unfortunately these will show up as a second, alphabetically sorted list
+	// of flags immediately concatenated with the above set.
+	upbound.Flags `embed:""`
+
+	// Internal state. These aren't part of the user-exposed CLI structure.
 	stdin  *os.File
 	client *http.Client
+}
```

Comment:
This renders as:

```
$ go run cmd/crank/main.go xpkg login --help
Usage: crossplane xpkg login

Login to the default package registry.

Crossplane can be extended using packages. A Crossplane package is sometimes
called an xpkg. Crossplane supports configuration, provider and function
packages.

A package is an opinionated OCI image that contains everything needed to extend
Crossplane with new functionality. For example installing a provider package
extends Crossplane with support for new kinds of managed resource (MR).

This command logs in to the xpkg.upbound.io package registry. The Crossplane CLI
pushes packages to xpkg.upbound.io by default.

See https://docs.crossplane.io/latest/concepts/packages for more information.

Flags:
  -h, --help               Show context-sensitive help.
      --verbose            Print verbose logging statements.
  -v, --version            Print version and quit.

  -p, --password=STRING    Password for specified username. '-' to read from
                           stdin ($UP_PASSWORD).
  -t, --token=STRING       Token used to authenticate. '-' to read from stdin
                           ($UP_TOKEN).
  -u, --username=STRING    Username used to authenticate ($UP_USER).
  -a, --account=STRING     Account used to execute command ($UP_ACCOUNT).
      --domain=https://upbound.io
                           Root Upbound domain ($UP_DOMAIN).
      --insecure-skip-tls-verify
                           [INSECURE] Skip verifying TLS certificates
                           ($UP_INSECURE_SKIP_TLS_VERIFY).
      --profile=STRING     Profile used to execute command ($UP_PROFILE).
```

Note that the `'-' to read from stdin` does actually mean that. i.e. You can do `echo my-username|crossplane xpkg login --username=-`. I think its safe to assume that most folks who will be using this CLI know what stdin is.

Reply:
Also, I don't love the "used to execute command" wording on some of these. Unfortunately they're shared flags intended to be embedded across a number of subcommands so it's tough to be more specific about what they're for exactly.